src/Namshi/JOSE/JWS.php

```diff
@@ -94,9 +104,9 @@ public static function load($jwsTokenString)
             $payload = json_decode($encoder->decode($parts[1]), true);
 
             if (is_array($header) && is_array($payload)) {
-                $jws = new self($header['alg'], isset($header['typ']) ? $header['typ'] : null);
+                $jws = new self($header);
                 $jws->setEncoder($encoder);
-                $jws->setPayload($payload);
+                $jws->setPayload($payload, false);
```

Comment:
That should be fine, because they will be in the payload already, and the check that was already in setPayload to not reset them is still there (which is why it worked at all).